pkg/controller/storagenode/storagenode.go

```diff
@@ -267,14 +278,14 @@ func (c *Controller) syncKVDB(
 			isRecentlyCreatedAfterNodeCordoned = k8s.IsPodRecentlyCreatedAfterNodeCordoned(node, c.nodeInfoMap, cluster)
 		}
 
-		if !isBeingDeleted && !isRecentlyCreatedAfterNodeCordoned && len(kvdbPodList.Items) == 0 {
+		if !isBeingDeleted && !isRecentlyCreatedAfterNodeCordoned && len(kvdbPods) == 0 {
 			pod, err := c.createKVDBPod(cluster, storageNode)
 			if err != nil {
 				return err
 			}
 
 			c.log(storageNode).Infof("creating kvdb pod: %s/%s", pod.Namespace, pod.Name)
-			err = c.client.Create(context.TODO(), pod)
+			_, err = coreops.Instance().CreatePod(pod)
```

Comment:
I think it's safe to use the controller client API here and below as well? do we need to replace all of them?